### PR TITLE
Feature/use fl slot set methods

### PR DESCRIPTION
**Pull request by Sandychuang8**
CBL-650 
Part 1: Fix tests in ArrayTest and CSharpTest mainly by updating current way of value setting in FLMutableArray and converting FLValue from FLSliceResult using FLDoc API
Part 2: Fix Predictive query tests with AggregateModel (https://github.com/couchbase/couchbase-lite-net/pull/1204)
Part 3: Replication tests (https://github.com/couchbase/couchbase-lite-net/pull/1205)

### CODE REVIEW

File: src/LiteCore/parse/templates_c4/C4ReplicatorParameters.cs [hunk 1]
```diff
@@ -11,13 +11,4 @@ internal unsafe partial struct C4ReplicatorParameters
         public void* callbackContext;
         public C4SocketFactory* socketFactory;
         private byte _dontStart;
```
Comment (borrrden) [inline src/LiteCore/parse/templates_c4/C4ReplicatorParameters.cs]: If `dontStart` is removed `_dontStart` should go with it

File: src/LiteCore/parse/templates_c4/C4SocketFactory.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-    internal unsafe struct C4SocketFactory
+    internal unsafe partial struct C4SocketFactory
```
Comment (borrrden) [inline src/LiteCore/parse/templates_c4/C4SocketFactory.cs]: What is the purpose of making this partial (and the others that were made partial)?

Comment (Sandychuang8) [reply src/LiteCore/parse/templates_c4/C4SocketFactory.cs]: I thought they are generated by binding script?

Comment (borrrden) [reply src/LiteCore/parse/templates_c4/C4SocketFactory.cs]: This one is a template, and the binding script configuration was also modified as part of this PR.  I'm just wondering if there is a reason to change them.

Comment (Sandychuang8) [reply src/LiteCore/parse/templates_c4/C4SocketFactory.cs]: There is no reason to change the script anymore. Let me try to revert them all.

**Pull request merged**